src/porepy/models/compositional_flow.py

```diff
@@ -0,0 +1,2070 @@
+  :class:`BoundaryConditionsPhaseProperties` for the fractional flow setting, where
+  various non-linear terms in fluxes can be given explicitly on the boundary,
+  instead of providing variable values which in return compute phase properties
+  appearing in those expressions.
+- :class:`BoundaryConditionsFlowAndTransport`: A collection of BC update routines for
+  primary variables and phase properties as surrogate factories, including those from
+  mass & energy, enthalpy and fractions.
+- :class:`BoundaryConditionsFractionalFlowAndTransport`: The alternative to
+  :class:`BoundaryConditionsFlowAndTransport` for the fractional flow setting with
+  explicit values for non-linear weights in fluxes.
+
+The :class:`SolutionStrategyCF` handles the general Cf model, with or without fractional
+flow simulation, and provides means to re-discretize the MPFA in the diffusive setting,
+and to eliminate local, secondary equations via Schur-complement. Those equations are
+required in any case to close the general multi-phase setting.
+
+The two setups, :class:`ModelSetupCF` and :class:`ModelSetupCFF` are in principle
+complete set-ups for non-isothermal, compositional flow. The steps required by users to
+close the setups are:
+
+1. Define a fluid with all its phases and components.
+2. Close the system with local equations for dangling, fractional variables. These can
+   either be :class:`~porepy.models.abstract_equations.LocalElimination` using some
+   map or third-party correlations, or a closure in form of a local equilibrium system.
+
+"""
+
+from __future__ import annotations
+
+import logging
+import time
+from functools import partial
+from typing import Callable, Optional, Sequence, cast
+
+import numpy as np
+
+import porepy as pp
+import porepy.compositional as ppc
+
+from .protocol import CompositionalFlowModelProtocol
+
+logger = logging.getLogger(__name__)
+
+
+def update_phase_properties(
+    grid: pp.Grid,
+    phase: pp.Phase,
+    props: ppc.PhaseProperties,
+    depth: int,
+    update_derivatives: bool = True,
+    use_extended_derivatives: bool = False,
+) -> None:
+    """Helper method to update the phase properties and its derivatives.
+
+    This method is intended for a grid-local update of properties and their derivatives,
+    using the methods of the
+    :class:`~porepy.numerics.ad.surrogate_operator.SurrogateFactory`.
+
+    Parameters:
+        grid: A grid in the md-domain.
+        phase: Any phase.
+        props: A phase property structure containing the new values to be set.
+        depth: An integer used to shift existing iterate values backwards.
+        update_derivatives: ``default=True``
+
+            If True, updates also the derivative values.
+        use_extended_derivatives: ``default=False``
+
+            If True, and if ``update_derivatives==True``, uses the extend
+            derivatives of the ``state``.
+
+            To be used in the the CFLE setting with the unified equilibrium formulation.
```

Comment:
There are explanations for those abbreviations in the module documentation. I added one for CFLE.
It's not a glossary, but in-text. I think it's fine at this stage, because it's only CF, CFF, CFLE.